### ISSUE TITLE
Make ColumnDef fields public

### ISSUE THREAD

**Issue by oscartbeaumont** [SeaQL/sea-orm#317]
Hello,
I am creating a package called [sea-migrations](https://github.com/oscartbeaumont/sea-migrations) which handles doing automatic database migrations for sea-orm. To do database migrations I require information from sea-orm about the desired structure of the database.

I found that the relationship data (`RelationDef`) [defined here](https://github.com/SeaQL/sea-orm/blob/d5da351f50ab44aaef2104ad1450f15c0b93c8e9/src/entity/relation.rs#L46) in the sea-orm package has all of its fields public.

However, the column data (`ColumnDef`) [defined here](https://github.com/SeaQL/sea-query/blob/master/src/table/column.rs#L5) in sea-query does not expose its fields publically.

Would it be possible for these fields to be changed to be public similar to the relationship data? Alternatively could getter methods be implemented to fetch the data stored within the struct?

I would be able to do a pull request if desired, I just wanted to check first.

**Comment by oscartbeaumont** [comment]
I gave fixing this issue an attempt by removing the ColumnDef defined by `sea-orm` and upgrading sea-query's `ColumnDef` to satisfy all the required functionality.

Doing this introduces a large number of `.unwrap()` calls due to sea-queries `ColumnDef.get_column_type()` method returning an `Option<&ColumnType>` as opposed to `&ColumnType` directly like sea-orm's `ColumnDef` does.

Another issue I encountered with this solution is that [here](https://github.com/oscartbeaumont/sea-orm/blob/master/src/entity/column.rs#L294) I can't determine the `Iden` of the Column which is required for `sea-migrations`. But I'm not sure that is the best solution.

I would be interested to hear if you think this solution is worth pursuing or if you think there is a better way, I am obviously not super familiar with the codebase so I could be missing a better method.

Another solution would be to add getters/setter functions to the `impl ColumnDef` in SeaORM but I also wanted to implement `From<ColumnDef>` for `sea_query::ColumnDef`and that's what lead to my current attempt at a solution as the sea_query `ColumnDef`  always required the `Iden` of the column.

The code for my attempt can be found [here (sea-query)](https://github.com/oscartbeaumont/sea-query/blob/f61a2bb89cb5b44bf019b89e19cd4667a167ccbc/src/table/column.rs) and [here (sea-orm)](https://github.com/oscartbeaumont/sea-orm).

**Comment by tyt2y3** [comment]
Yeah, it's a bit unfortunate that there is mismatch between sea-query and sea-orm, and its because they were catering different needs in the first place.

I agree, unifying them would lead to less confusion. If we go down this path, some features have to be removed from sea-query, to make it less 'raw'.

**Comment by tyt2y3** [comment]
I agree on pursuing this direction, it's a debt we have to pay at some point.

@billy1624 thoughts?

**Comment by billy1624** [comment]
I agree on unifying the `ColumnType ` & `ColumnDef` but I think the "Raw SQL" feature such as column width of integers should be achieve in other ways in `sea-query`. For example, using a wrapper struct like below.

```rust
pub struct ColumnDef {
    pub(crate) table: Option<TableRef>,
    pub(crate) name: DynIden,
    pub(crate) types: Option<ColumnType>,
    pub(crate) spec: Vec<ColumnSpec>,
}

pub struct Column {
    pub(crate) col_def: ColumnDef,
    pub(crate) length:  u32,
}
```

**Comment by tyt2y3** [comment]
If this issue still relevant, please reopen
